### ISSUE TITLE
Strange behavior for specific key names

### ISSUE THREAD

**Issue by ghost** [starapor/slippers#1]
This works for me
template = Slippers::Engine.new("SELECT $column$ FROM $table$;")
puts template.render(:column => "name", :table => "User") === "SELECT name FROM User;"

 but if i change the name of the a key to "gender" like this

template = Slippers::Engine.new("SELECT $gender$ FROM $table$;")
puts template.render(:gender=> "name", :table => "User") === ""

render returns nil.  This bug manifests itself for the strings gender, gende, gend but gen works ok.

Very curious....

Thanks

**Comment by starapor** [comment]
Yeah - you're right! I have been able to reproduce this in my tests and found the cause - it is because of the "end" in the middle there. End is a keyword because of $if$ .. $end$ but it shoudn't be picking it up on words like gender.
I am looking into this now. Incidentally did you find this because you just played around, or are you using it in your project?
Cheers
Sarah

**Comment by ghost** [comment]
No this is part of a larger project of populating a clinial document architecture (XML) document with data from an electronic medical record hence the need for "gender".  I am doing this on my own and learning Ruby simultaenously.   BTW I think i also found an issue on your slippers vs stringtemplate page.  I think that under "Loading Templates from a File" that 

template_group = Slippers::TemplateGroupDirectory.new(['tmp'])
query = template_group.find('theQuery')
st = Slippers::Engine.new(template_group)
st.render(:column => ["name", "email"], :table => "User")

it should be query.render rather than st.render

but I could be wrong.  Thanks for responding so quickly, I really do like the library and the philosophy that guided you.

Best

Barry
